### PR TITLE
Set use_develop in tox.ini

### PR DESCRIPTION
**Pull request by pabelanger**
We are using ansible to install navigator into tox, we do not need to do
this. Lets set use_develop and have tox to it for free.

Signed-off-by: Paul Belanger <pabelanger@redhat.com>

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -34,11 +33,11 @@ setenv =
 passenv = HOME
           USER
           XDIST_CPU
+usedevelop = True
 
 [testenv:clean]
 description = Erase coverage data
 deps = coverage
-skip_install = true
```
Comment (webknjaz) [inline tox.ini:41]: Don't do this. This env doesn't need navigator at all.

File: tox.ini [hunk 2]
```diff
@@ -10,7 +10,6 @@ minversion = 1.4.2
 skipsdist = True
 skip_missing_interpreters = true
```
Comment (webknjaz) [inline tox.ini:12]:
```suggestion


```

File: tox.ini [hunk 3]
```diff
@@ -61,13 +63,14 @@ commands =
 
 # Note included in the default envlist above since it's destructive
 # (requires tmux, git, runs a bunch of commands, etc.)
+
 [testenv:smoke]
 description = Run smoke tests under {basepython} ({envpython})
 commands = ansible-playbook tests/smoketests/run.yml
-allowlist_externals = ansible-playbook
```
Comment (webknjaz) [inline tox.ini:67]: This needs to be recovered.

File: tox.ini [hunk 4]
```diff
@@ -113,10 +116,8 @@ commands =
 isolated_build = true
 passenv =
   SSH_AUTH_SOCK
-skip_install = false
 usedevelop = true
```
Comment (webknjaz) [inline tox.ini:120]:
```suggestion


```

File: tox.ini [hunk 5]
```diff
@@ -61,13 +63,14 @@ commands =
...
 # (requires tmux, git, runs a bunch of commands, etc.)
+
 [testenv:smoke]
 description = Run smoke tests under {basepython} ({envpython})
 commands = ansible-playbook tests/smoketests/run.yml
-allowlist_externals = ansible-playbook
 
 [testenv:type]
 description = Verify static typing with MyPy under {basepython} ({envpython})
+skip_install = false
```
Comment (ssbarnea) [inline tox.ini:73]: is this really needed? seem unrelated change

File: tox.ini [hunk 6]
```diff
@@ -113,7 +116,6 @@ commands =
 isolated_build = true
 passenv =
   SSH_AUTH_SOCK
-skip_install = false
```
Comment (ssbarnea) [inline tox.ini:116]: also seems unrelated.

**Review by ssbarnea — CHANGES_REQUESTED** [review]

**Comment by cidrblock** [comment]
@ssbarnea @webknjaz Givent he changes in the repo recently is this still needed?

**Comment by ssbarnea** [comment]
There are several problems with it, one of them being that it combines at least 3 different changes. My guess is that the standalone use of pip outside tox with smoke was intended and not an accident as the intent was to see if installation with pip works. What tox does is not always the same thing as pip, especially when using develop. Also testing install outside a virtualenv is desirable as if we use only tox venvs we may fail to identify other bugs, like conflicts with system python packages.

So, I think that PR is no longer needed and I will close it but we can reopen it at any time if needed.

**Pull request closed without merging**